.github/workflows/release.yml

```
@@ -0,0 +1,65 @@
+name: Release When Tagged
+run-name: Release ${{ github.ref_name }}
+
+on:
+  push:
+    tags: '*'
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    container:
+      image: rancher/dapper:v0.6.0
+    permissions:
+      contents: read
+    steps:
+      - name: Check For Alpha / RC in Release
+        if: contains(github.ref_name, 'rc')  || contains(github.ref_name, 'alpha')
+        uses: actions/github-script@v7
+        with:
+          script: |
+            core.setFailed('No RCs or Alphas are released in this repo. Skipping release...') 
+
+      - name: Fix the not-a-git-repository issue
+        run: |
+          apk -U add git
+          git config --global --add safe.directory "$GITHUB_WORKSPACE"
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+
+      - name: build with Dapper
+        run: dapper ci
+        
+      - name: Archive production artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: kubernetes-build
+          path: output/bin
+          if-no-files-found: error
+          overwrite: true
+
+  create-release:
+    runs-on: ubuntu-latest
+    needs:
+      - validate
+    permissions:
+      contents: write # needed for creating the GH release
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

GH_TOKEN is the recommended env var to use according to https://docs.github.com/en/actions/using-workflows/using-github-cli-in-workflows
```suggestion
      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

---

updated to GH_TOKEN